mmv1/products/firestore/Field.yaml

```diff
@@ -0,0 +1,161 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'Field'
+base_url: projects/{{project}}/databases/{{database}}/collectionGroups/{{collection}}/fields
+create_url: projects/{{project}}/databases/{{database}}/collectionGroups/{{collection}}/fields/{{field}}
+self_link: '{{name}}'
+immutable: false
+update_verb: :PATCH
+update_mask: true
+create_verb: :PATCH
+description: |
+  Represents a single field in the database.
+  Fields are grouped by their "Collection Group", which represent all collections
+  in the database with the same id.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/firestore/docs/query-data/indexing'
+  api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.collectionGroups.fields'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+skip_sweeper: true
+import_format: ["{{name}}"]
+docs: !ruby/object:Provider::Terraform::Docs
+  warning: |
+    This resource creates a Firestore Single Field override on a project that
+     already has a Firestore database. If you haven't already created it, you may
+    create a `google_firestore_database` resource with `location_id` set to your
+    chosen location.
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firestore_field_basic"
+    primary_resource_id: "basic"
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firestore_field_timestamp"
+    primary_resource_id: "timestamp"
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+  - !ruby/object:Provider::Terraform::Examples
+    name: "firestore_field_match_override"
+    primary_resource_id: "match_override"
+    test_env_vars:
+      project_id: :FIRESTORE_PROJECT_NAME
+custom_code: !ruby/object:Provider::Terraform::CustomCode
+  custom_import: templates/terraform/custom_import/index_self_link_as_name_set_project.go.erb
+  encoder: templates/terraform/encoders/firestore_field.go.erb
+  custom_delete: templates/terraform/custom_delete/firestore_field_delete.go.erb
+  test_check_destroy: templates/terraform/custom_check_destroy/firestore_field.go.erb
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'database'
+    default_value: '(default)'
```

Comment:
I was going to ask about this default value, and whether there should be an acceptance tests that sets `database` to another value, but then I saw this [in the docs](https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases#resource:-database):

>A Cloud Firestore Database. Currently only one database is allowed per cloud project; this database must have a databaseId of '(default)'.

Noting this on the PR partly for myself as I'm unfamiliar with Firestore, partly for the benefit of anyone looking at the PR in future!

Reply:
👍  Once this restriction changes, I expect we should use generative ones for testing.